src/python_testing/TC_SC_3_6.py

```diff
@@ -0,0 +1,226 @@
+                new_controllers = await CommissioningBuildingBlocks.CreateControllersOnFabric(fabricAdmin=new_fabric_admin, adminDevCtrl=new_admin_ctrl,
+                                                                                              controllerNodeIds=node_ids, privilege=Clusters.AccessControl.Enums.Privilege.kAdminister, targetNodeId=self.dut_node_id)
+                for controller in new_controllers:
+                    controller.name = all_names.pop(0)
+
+                client_list.extend(new_controllers)
+
+        asserts.assert_equal(len(client_list), num_fabrics_to_commission *
+                             num_controllers_per_fabric, "Must have the right number of clients")
+
+        # Before subscribing, set the NodeBabel to "Before Subscriptions"
+        logging.info("Pre-conditions: writing initial value of NodeLabel, so that we can control for change of attribute detection")
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=BEFORE_LABEL))])
+
+        # Subscribe with all clients to NodeLabel attribute
+        subscriptions = []
+        sub_handlers = []
+        resub_catchers = []
+
+        logging.info("Step 1 (first part): Establish subscription with all 15 clients")
+        for client in client_list:
+            logging.info("Establishing subscription from controller node %s" % client.name)
+            sub = await client.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, Clusters.Basic.Attributes.NodeLabel)],
+                                             reportInterval=(min_report_interval_sec, max_report_interval_sec), keepSubscriptions=False)
+            subscriptions.append(sub)
+
+            attribute_handler = AttributeChangeAccumulator(name=client.name)
+            sub.SetAttributeUpdateCallback(attribute_handler)
+            sub_handlers.append(attribute_handler)
+
+            resub_catcher = ResubscriptionCatcher(name=client.name)
+            sub.SetResubscriptionAttemptedCallback(resub_catcher, isAsync=True)
+            resub_catchers.append(resub_catcher)
+
+            if sub_liveness_override_ms is not None:
+                logging.warning("Overriding subscription liveness to check %dms! NOT FOR CERTIFICATION!" % sub_liveness_override_ms)
+                sub.OverrideLivenessTimeoutMs(sub_liveness_override_ms)
+
+        asserts.assert_equal(len(subscriptions), num_fabrics_to_commission *
+                             num_controllers_per_fabric, "Must have the right number of subscriptions")
+
+        # Trigger a change on NodeLabel
+        logging.info("Step 1 (second part): Change attribute with one client, and validate all clients observe the change on same session")
+        await asyncio.sleep(1)
+        await client_list[0].WriteAttribute(self.dut_node_id, [(0, Clusters.Basic.Attributes.NodeLabel(value=AFTER_LABEL))])
+
+        # Await a stabilization delay in increments to let the coroutines run
```

Comment:
Why in increments?

Reply:
So, I was seeing that this sleep, if left full prevented all the sub callbacks from firing. It turns out `handleReportEnd` is not async, so that's why nothing was working otherwise. In fact, I think there is a concurrency issue in handling of subs between async and non-async code :(

```
def handleReportEnd(self):
        # self._event_loop.call_soon_threadsafe(self._handleReportEnd)
        self._handleReportEnd()
```

Reply:
Fixed per offline discussion